src/Monolog/Attribute/WithMonologChannel.php

```diff
@@ -0,0 +1,29 @@
+<?php declare(strict_types=1);
+
+/*
+ * This file is part of the Monolog package.
+ *
+ * (c) Jordi Boggiano <j.boggiano@seld.be>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Monolog\Attribute;
+
+/**
+ * A reusable attribute to help configure a class as expecting a given logger channel.
+ *
+ * Using it offers no guarantee: it needs to be leveraged by a Monolog third-party consumer.
+ *
+ * Using it with the Monolog library only has no effect at all: wiring the logger instance into
+ * other classes is not managed by Monolog.
+ */
+#[\Attribute(\Attribute::TARGET_CLASS)]
```

Comment:
Should be allow it on parameters also, and possibly properties?

Reply:
This won't work in MonologBundle. The tag would apply to the whole service. It does not support separate tags per parameter.
If you want to inject 2 different loggers in the same class with autowiring, you will need to rely on the named aliases or `#[Target]`

Reply:
Couldn't we turn the attribute into eg a binding in the bundle?

Reply:
btw, that's also why I did not make the attribute repeatable.

Reply:
binding are not per-attribute but per service.